cpp/unreal_plugins/SpEngine/Source/SpEngine/GameWorldService.h

```diff
@@ -36,6 +39,107 @@ struct GameWorldServicePropertyDesc
+
+    /** Method for resolving collisions at the spawn point. Undefined means no override, use the actor's setting. */
+    ESpawnActorCollisionHandlingMethod SpawnCollisionHandlingOverride;
+
+    /** Determines whether to multiply or override root component with provided spawn transform */
+    ESpawnActorScaleMethod TransformScaleMethod = ESpawnActorScaleMethod::MultiplyWithRoot;
+
+private:
+
+    friend class UPackageMapClient;
+
+#if UE_WITH_IRIS
+    friend class UActorReplicationBridge;
+#endif // UE_WITH_IRIS
+
+    /* Is the actor remotely owned. This should only be set true by the package map when it is creating an actor on a client that was replicated from the server. */
+    uint8	bRemoteOwned : 1;
+
+public:
+
+    bool IsRemoteOwned() const { return bRemoteOwned; }
+
+    /* Determines whether spawning will not fail if certain conditions are not met. If true, spawning will not fail because the class being spawned is `bStatic=true` or because the class of the template Actor is not the same as the class of the Actor being spawned. */
+    uint8	bNoFail : 1;
+
+    /* Determines whether the construction script will be run. If true, the construction script will not be run on the spawned Actor. Only applicable if the Actor is being spawned from a Blueprint. */
+    uint8	bDeferConstruction : 1;
+
+    /* Determines whether or not the actor may be spawned when running a construction script. If true spawning will fail if a construction script is being run. */
+    uint8	bAllowDuringConstructionScript : 1;
+
+#if WITH_EDITOR
+    /* Determines whether the begin play cycle will run on the spawned actor when in the editor. */
+    uint8	bTemporaryEditorActor : 1;
+
+    /* Determines whether or not the actor should be hidden from the Scene Outliner */
+    uint8	bHideFromSceneOutliner : 1;
+
+    /** Determines whether to create a new package for the actor or not, if the level supports it. */
+    uint16	bCreateActorPackage : 1;
+#endif
+
+    /* Modes that SpawnActor can use the supplied name when it is not None. */
+    enum class ESpawnActorNameMode : uint8
+    {
+        /* Fatal if unavailable, application will assert */
+        Required_Fatal,
+
+        /* Report an error return null if unavailable */
+        Required_ErrorAndReturnNull,
+
+        /* Return null if unavailable */
+        Required_ReturnNull,
+
+        /* If the supplied Name is already in use the generate an unused one using the supplied version as a base */
+        Requested
+    };
+
+    /* In which way should SpawnActor should treat the supplied Name if not none. */
+    ESpawnActorNameMode NameMode;
+
+    /* Flags used to describe the spawned actor/object instance. */
+    EObjectFlags ObjectFlags;
+
+    /* Custom function allowing the caller to specific a function to execute post actor construction but before other systems see this actor spawn. */
+    TFunction<void(AActor*)> CustomPreSpawnInitalization;
+};
+
```

Comment:
I think we can get rid of `set_game_paused` and `open_level` at this point, because a user can accomplish the same result with `get_static_class`, `get_default_object`, and `call_function`.

Reply:
Updated code to do this. It's a bit verbose, but gives a perfect example of how users can compose different function calls to achieve what they want.

Reply:
100% agree. You're right that this new style can be bit more verbose on the Python side, and you're right that it'll be a useful reference for users (and ourselves).

For what it's worth, I think some of the added verboseness is unavoidable, given that we're intentionally trying to make our Python interface lower-level, such that it maps more directly to Unreal function calls.